src/mylib/common.cpp

```diff
@@ -21,14 +21,17 @@
  *
  */
 
+#include <mylib/common.h>
+
 #include <geode/basic/common.h>
 
 namespace
 {
-    OPENGEODE_LIBRARY_INITIALIZE( myLib )
+    OPENGEODE_LIBRARY_IMPLEMENTATION( MyModuleMyLib )
     {
         /* Here the functions to call when initializing the library
          * For exemple: registers, ...
          */
+        MyModuleMyLib::initialize();
```

Comment:
I think the template depends on OpenGeodeasic::initialize() too

Reply:
This is not clear to me @BotellaA .
Should we initialize every dependent libraries. 
for example in [the opengeode-gesciences ](https://github.com/Geode-solutions/OpenGeode-Geosciences/commit/bd18bae4ae48ff7db14006b089f347bba6157d1f#:~:text=OPENGEODE_LIBRARY_INITIALIZE(%20geosciences%20)-,namespace%20geode,%7D,-%7D%20//%20namespace
 ) project you wrote: : 
`    OPENGEODE_LIBRARY_IMPLEMENTATION( OpenGeodeGeosciencesGeosciences )
    {
        OpenGeodeModel::initialize();
        register_cross_section_input();
        register_cross_section_output();
        register_structural_model_input();
        register_structural_model_output();
    }`
but this project depend on both  OpenGeodeModel and OpenGeodeBasic..
Also here you called MyModuleMyLib::initialize() but in the  opengeode-gesciences project you did not called:
        OpenGeodeGeosciencesGeosciences::initialize();
could you clarify that to me please.

Reply:
It is a dependence tree. Each node will initialize its parent nodes.

Then in the executable, you only need to initialize the leaves of the tree. But it will also work if you initialize all of them, this is just not necessary.